pkg/autoscaler/k8sclient/k8sclient.go

```diff
@@ -226,39 +254,39 @@ func (k *k8sClient) UpdateReplicas(expReplicas int32) (prevRelicas int32, err er
 		glog.V(0).Infof("Cluster status: SchedulableNodes[%v], TotalNodes[%v], SchedulableCores[%v], TotalCores[%v]", k.clusterStatus.SchedulableNodes, k.clusterStatus.TotalNodes, k.clusterStatus.SchedulableCores, k.clusterStatus.TotalCores)
 		glog.V(0).Infof("Replicas are not as expected : updating replicas from %d to %d", prevRelicas, expReplicas)
 		scale.Spec.Replicas = expReplicas
-		_, err = k.updateScaleExtensionsV1beta1(k.target, scale)
+		_, err = k.updateScaleExtensionsV1beta1(&target, namespace, scale)
 		if err != nil {
 			return 0, err
 		}
 	}
 	return prevRelicas, nil
 }
 
-func (k *k8sClient) getScaleExtensionsV1beta1(target *scaleTarget) (*extensionsv1beta1.Scale, error) {
+func (k *k8sClient) getScaleExtensionsV1beta1(target *target, namespace string) (*extensionsv1beta1.Scale, error) {
 	opt := metav1.GetOptions{}
 	switch strings.ToLower(target.kind) {
 	case "deployment", "deployments":
-		return k.clientset.ExtensionsV1beta1().Deployments(target.namespace).GetScale(context.TODO(), target.name, opt)
+		return k.clientset.ExtensionsV1beta1().Deployments(namespace).GetScale(context.TODO(), target.name, opt)
 	case "replicaset", "replicasets":
-		return k.clientset.ExtensionsV1beta1().ReplicaSets(target.namespace).GetScale(context.TODO(), target.name, opt)
+		return k.clientset.ExtensionsV1beta1().ReplicaSets(namespace).GetScale(context.TODO(), target.name, opt)
 	default:
 		return nil, fmt.Errorf("unsupported target kind: %v", target.kind)
 	}
 }
 
-func (k *k8sClient) updateScaleExtensionsV1beta1(target *scaleTarget, scale *extensionsv1beta1.Scale) (*extensionsv1beta1.Scale, error) {
+func (k *k8sClient) updateScaleExtensionsV1beta1(target *target, namespace string, scale *extensionsv1beta1.Scale) (*extensionsv1beta1.Scale, error) {
 	switch strings.ToLower(target.kind) {
 	case "deployment", "deployments":
-		return k.clientset.ExtensionsV1beta1().Deployments(target.namespace).UpdateScale(context.TODO(), target.name, scale, metav1.UpdateOptions{})
+		return k.clientset.ExtensionsV1beta1().Deployments(namespace).UpdateScale(context.TODO(), target.name, scale, metav1.UpdateOptions{})
 	case "replicaset", "replicasets":
-		return k.clientset.ExtensionsV1beta1().ReplicaSets(target.namespace).UpdateScale(context.TODO(), target.name, scale, metav1.UpdateOptions{})
+		return k.clientset.ExtensionsV1beta1().ReplicaSets(namespace).UpdateScale(context.TODO(), target.name, scale, metav1.UpdateOptions{})
 	default:
 		return nil, fmt.Errorf("unsupported target kind: %v", target.kind)
 	}
 }
 
-func (k *k8sClient) updateReplicasAppsV1(expReplicas int32) (prevRelicas int32, err error) {
-	req, err := requestForTarget(k.clientset.AppsV1().RESTClient().Get(), k.target)
+func (k *k8sClient) updateReplicasAppsV1(expReplicas int32, target target, namespace string) (prevRelicas int32, err error) {
```

Comment:
same, no need for namespace as you already have it in `k. scaleTargets.namespace`.
Also, is it possible to use pointer for `target` here?

Reply:
I passed the target by value to show the function is not mutating the parameters, but I can change it to a pointer if you think it's more consistent with the rest of the code